celery/app/trace.py

```diff
@@ -364,10 +364,15 @@ def trace_task(uuid, args, kwargs, request=None):
 
                 # -*- TRACE -*-
                 try:
-                    from celery.concurrency.future import future_executor
-                    R = retval = future_executor.apply_future(
-                        fun, *args, **kwargs)
-                    state = SUCCESS
+                    try:
+                        from celery.concurrency.future import get_future_executor
```

Comment:
In this case, the default future executor will do the right job as before. If the task is a coroutine function, which will be partially executed in the tornado ioloop. But the result of coroutine function can't immediately return.